### PR TITLE
Add MapLayers enum.

### PR DESCRIPTION
**Pull request by siarheishut**
Add enum for maintaining layer types.

### CODE REVIEW

File: src/request_types.h [hunk 1]
```diff
@@ -19,6 +19,13 @@ struct RoutingSettings {
   double bus_velocity;
 };
 
+enum MapLayers {
```
Comment (foxfrikses) [inline src/request_types.h:22]: 1. enum class
2. MapLayer
3. kBusRoutes, ...

File: tests/settings_parser_test.cpp [hunk 2]
```diff
@@ -82,7 +82,9 @@ TEST(TestParseSettings, TestRedneringSettings) {
                                         json::List{210, 81, 14, 0.94}}},
                                     {"bus_label_font_size", 6},
                                     {"bus_label_offset", json::List{5, -7}},
-                                    {"layers", json::List{"1", "2", "3"}}};
+                                    {"layers", json::List{"bus_labels",
+                                                          "stop_points",
```
Comment (foxfrikses) [inline tests/settings_parser_test.cpp:86]: Add tests for all the types of layers + multiple invalid strings + for empty layers array

File: src/request_types.h (hunk 1 above)
Comment (siarheishut) [reply src/request_types.h]: Done.

File: tests/settings_parser_test.cpp (hunk 2 above)
Comment (siarheishut) [reply tests/settings_parser_test.cpp:86]: Done.

File: src/request_parser.cpp [hunk 3]
```diff
@@ -46,11 +46,20 @@ bool IsLayers(const json::Node &node) {
...
 }
 
-std::vector<std::string> AsLayers(json::Node node) {
-  std::vector<std::string> layers;
+std::vector<rm::MapLayer> AsLayers(json::Node node) {
+  std::vector<rm::MapLayer> layers;
   layers.reserve(node.AsArray().size());
   for (auto &item : node.ReleaseArray()) {
-    layers.push_back(item.ReleaseString());
+    auto layer = item.ReleaseString();
```
Comment (foxfrikses) [inline src/request_parser.cpp:53]: Why release? Why not use refs?

File: src/request_parser.cpp [hunk 4]
```diff
@@ -46,11 +46,20 @@ bool IsLayers(const json::Node &node) {
       });
 }
 
-std::vector<std::string> AsLayers(json::Node node) {
-  std::vector<std::string> layers;
+std::vector<rm::MapLayer> AsLayers(json::Node node) {
```
Comment (foxfrikses) [inline src/request_parser.cpp:49]: you can use const ref here

File: src/request_parser.cpp (hunk 3 above)
Comment (siarheishut) [reply src/request_parser.cpp]: Done.

File: src/request_parser.cpp (hunk 4 above)
Comment (siarheishut) [reply src/request_parser.cpp]: Done.

File: tests/settings_parser_test.cpp [hunk 5]
```diff
@@ -93,6 +96,31 @@ TEST(TestParseSettings, TestRedneringSettings) {
...
+          svg::Rgba{.red = 210, .green = 81, .blue = 14, .alpha = 0.94}},
+      .bus_label_font_size = 6,
+      .bus_label_offset = {.x = 5, .y = -7},
+      .layers = {MapLayer::kBusNames,
+                 MapLayer::kStopPoints,
+                 MapLayer::kBusRoutes,
+                 MapLayer::kStopNames}
+  };
+
+  auto settings_layers_replace = [&](std::vector<MapLayer> layers) {
```
Comment (foxfrikses) [inline tests/settings_parser_test.cpp:118]: Maybe `want_item_replace_layers_with()`?

File: tests/settings_parser_test.cpp [hunk 6]
```diff
@@ -93,6 +96,31 @@ TEST(TestParseSettings, TestRedneringSettings) {
     item[field] = std::move(node);
     return item;
   };
+
+  const auto settings_item = RenderingSettings{
```
Comment (foxfrikses) [inline tests/settings_parser_test.cpp:100]: Mb `want_item`

File: src/request_types.h [hunk 7]
```diff
@@ -19,6 +19,13 @@ struct RoutingSettings {
   double bus_velocity;
 };
 
+enum class MapLayer {
+  kBusRoutes = 0,
+  kBusNames = 1,
```
Comment (foxfrikses) [inline src/request_types.h:24]: Why don't you name these enum constants in the same way as the input strings?
bus_labels -> kBusLabels
bus_lines -> kBusLines
...

File: tests/settings_parser_test.cpp [hunk 8]
```diff
@@ -241,23 +269,30 @@ TEST(TestParseSettings, TestRedneringSettings) {
...
+              "layers", json::List{"1", "2", "3", "2", "bus_labels"}),
+          .want = settings_layers_replace({MapLayer::kBusNames})
+      },
+      TestCase{
+          .name = "Repeated layers",
+          .input = test_item_replace(
+              "layers", json::List{"bus_names", "stop_names",
+                                   "bus_names", "stop_names"}),
+          .want = settings_layers_replace(
+              {MapLayer::kBusNames, MapLayer::kStopNames,
```
Comment (foxfrikses) [inline tests/settings_parser_test.cpp:289]: You don't have "bus_names" key

File: tests/settings_parser_test.cpp (hunk 6 above)
Comment (siarheishut) [reply tests/settings_parser_test.cpp:100]: Changed to settings.

File: tests/settings_parser_test.cpp (hunk 5 above)
Comment (siarheishut) [reply tests/settings_parser_test.cpp:118]: Changed to settings_replace_layers_with.

File: tests/settings_parser_test.cpp (hunk 8 above)
Comment (siarheishut) [reply tests/settings_parser_test.cpp:289]: Fixed.

File: src/request_types.h (hunk 7 above)
Comment (siarheishut) [reply src/request_types.h:24]: Done.

File: tests/settings_parser_test.cpp [hunk 9]
```diff
@@ -82,7 +82,10 @@ TEST(TestParseSettings, TestRedneringSettings) {
                                         json::List{210, 81, 14, 0.94}}},
                                     {"bus_label_font_size", 6},
                                     {"bus_label_offset", json::List{5, -7}},
-                                    {"layers", json::List{"1", "2", "3"}}};
+                                    {"layers", json::List{"bus_labels",
+                                                          "stop_points",
+                                                          "bus_routes",
```
Comment (foxfrikses) [inline tests/settings_parser_test.cpp:87]: There is no bus_routes key expected. Plus fix this and the comparison of settings

Comment (siarheishut) [reply tests/settings_parser_test.cpp]: Fixed.

**Review by foxfrikses — APPROVED** [review]
LGTM
